py/src/braintrust/logger.py

```diff
@@ -710,121 +669,103 @@ def summarize(summarize_scores=True, comparison_experiment_id=None):
 
-    return _state.current_logger.get()
+    return _state.current_logger
 
 
 def current_span() -> Span:
-    """Return the currently-active span for logging (set by `with *.start_span` or `braintrust.with_current`). If there is no active span, returns a no-op span object, which supports the same interface as spans but does no logging.
+    """Return the currently-active span for logging (set by running a span under a context manager). If there is no active span, returns a no-op span object, which supports the same interface as spans but does no logging.
 
     See `Span` for full details.
     """
 
     return _state.current_span.get()
 
 
-def start_span(name=None, span_attributes={}, start_time=None, set_current=None, **event) -> Span:
-    """Toplevel function for starting a span. It checks the following (in precedence order):
-    * Currently-active span
-    * Currently-active experiment
-    * Currently-active logger
-
-    and creates a span in the first one that is active. If none of these are active, it returns a no-op span object.
-
-    Unless a name is explicitly provided, the name of the span will be the name of the calling function, or "root" if no meaningful name can be determined.
-
-    We recommend running spans bound to a context manager (`with start_span`) to automatically mark them as current and ensure they are terminated. If you wish to start a span outside a callback, be sure to terminate it with `span.end()`.
-
-    See `Span.startSpan` for full details.
-    """
+def get_span_parent_object() -> Union["Logger", "Experiment", Span]:
+    """Mainly for internal use. Return the parent object for starting a span in a global context."""
 
-    name = name or get_caller_location()["caller_functionname"] or "root"
-    kwargs = dict(name=name, span_attributes=span_attributes, start_time=start_time, set_current=set_current, **event)
     parent_span = current_span()
     if parent_span != NOOP_SPAN:
-        return parent_span.start_span(**kwargs)
+        return parent_span
 
     experiment = current_experiment()
     if experiment:
-        return experiment.start_span(**kwargs)
+        return experiment
 
     logger = current_logger()
     if logger:
-        return logger.start_span(**kwargs)
+        return logger
 
     return NOOP_SPAN
 
 
-class _CurrentObjectWrapper:
-    """Context manager wrapper for marking an experiment as current."""
-
-    def __init__(self, object_cvar, object):
-        self.object_cvar = object_cvar
-        self.object = object
-
-    def __enter__(self):
-        self.context_token = self.object_cvar.set(self.object)
-
-    def __exit__(self, type, value, callback):
-        del type, value, callback
-
-        self.object_cvar.reset(self.context_token)
+def _try_log_input_output(span, f_args, f_kwargs, output):
+    log_kwargs = dict(input=dict(args=f_args, kwargs=f_kwargs), output=output)
```

Comment:
is there a way we can format it so that it's just always a dict with the arg names? I.e. a function like

```
def foo(a, b, **kwargs):
  return a + b + kwargs["foo"]
```

`foo(1, b=2, foo=3)` would log:

```
{
  "input": {"a": 1, "b": 2, "foo": 3},
  "output": 6
}
```